sklearn/cluster/_agglomerative.py

```diff
@@ -21,7 +21,8 @@
 from ..neighbors import DistanceMetric
 from ..neighbors._dist_metrics import METRIC_MAPPING
 
-from . import _hierarchical_fast as _hierarchical
+# mypy error: Module 'sklearn.cluster' has no attribute '_hierarchical_fast'
+from . import _hierarchical_fast as _hierarchical  # type: ignore
```

Comment:
Yeah, normally one currently has to write stub files for cython extensions since they don't expose annotations  (see e.g. https://github.com/scipy/scipy/pull/11936). I'm hoping this could be done automatically in the future though, since Cython should have all relevant information; possibly https://github.com/cython/cython/issues/2552

Reply:
When `mypy` runs on the source (before anything is built), my understanding is that there is no way for `mypy` to find the module. Is this correct?

Reply:
Yes, indeed. For most Cython imports errors will be ignored due to the `--ignore-missing-import` flag, however not when importing whole modules for some reason,
https://github.com/python/mypy/issues/8575#issuecomment-602198802

Reply:
Also I'm getting these errors locally with scikit-learn correctly build. E.g. without this change,
```
$ mypy --ignore-missing-imports sklearn/cluster/_agglomerative.py 
sklearn/cluster/_agglomerative.py:25: error: Module 'sklearn.cluster' has no attribute '_hierarchical_fast'
Found 1 error in 1 file (checked 1 source file)
$ mypy --ignore-missing-imports sklearn
Success: no issues found in 599 source files
```
I don't really understand why the full run succeeds..

Reply:
> I don't really understand why the full run succeeds..

Probably unrelated, but lately I've found that `git diff upstream/master -u -- "*.py" | flake8 --diff` passes on some branches while `build_tools/circle/linting.sh` (used by the CI) doesn't.

Reply:
> Probably unrelated, but lately I've found that git diff upstream/master -u -- "*.py" | flake8 --diff passes on some branches while build_tools/circle/linting.sh (used by the CI) doesn't.

Yes, linting the diff has issues https://github.com/scikit-learn/scikit-learn/issues/11336#issuecomment-613724473 and I would like to start dealing with it once we are done with the release.